### PR TITLE
SOLR-15182: Remove remaining apispec files

### PR DESCRIPTION
**Pull request by gerlowskija**
# Description

Solr is in the process of moving away from 'apispec' defined v2 APIs, and towards annotated-based APIs.  Only a few apispec files remain.
 
# Solution

This commit removes the remaining apispec definitions.  `core.system.blob.json` and `core.system.blob.upload.json` are converted into annotated Java classes, and emptySpec.json is removed altogether.

# Tests

See V2BlobApiMappingTest.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/api/ApiBag.java [hunk 1]
```diff
@@ -58,6 +35,18 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/api/ApiBag.java:40]: My oldest of friends, `java.util.*;` ;-)

File: solr/core/src/java/org/apache/solr/handler/BlobHandler.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import static org.apache.solr.common.params.CommonParams.SORT;
 import static org.apache.solr.common.params.CommonParams.VERSION;
 
+import com.google.common.collect.Lists;
```
Comment (epugh) [inline solr/core/src/java/org/apache/solr/handler/BlobHandler.java:25]: isn't there a PR to get away from this?

**Review by epugh — APPROVED** [review]
Couple of nits, LGTM.

File: solr/core/src/java/org/apache/solr/handler/BlobHandler.java (hunk 2 above)
Comment (gerlowskija) [reply solr/core/src/java/org/apache/solr/handler/BlobHandler.java:25]: Mm, that's news to me, I'll keep an eye out and stop using it if it goes through.

File: solr/core/src/java/org/apache/solr/api/ApiBag.java [hunk 3]
```diff
@@ -17,37 +17,14 @@
...
 import org.apache.solr.common.SolrException;
 import org.apache.solr.common.SpecProvider;
-import org.apache.solr.common.util.CommandOperation;
-import org.apache.solr.common.util.ContentStream;
-import org.apache.solr.common.util.JsonSchemaValidator;
-import org.apache.solr.common.util.NamedList;
-import org.apache.solr.common.util.PathTrie;
-import org.apache.solr.common.util.Utils;
-import org.apache.solr.common.util.ValidatingJsonMap;
+import org.apache.solr.common.util.*;
```
[the remark this answers is not in the document]
Comment (epugh) [reply solr/core/src/java/org/apache/solr/api/ApiBag.java]: We really need some sort of default intellij setup....   in olden days I think you did "ant intellij" and everything was set up!

File: solr/core/src/java/org/apache/solr/handler/BlobHandler.java (hunk 2 above)
Comment (epugh) [reply solr/core/src/java/org/apache/solr/handler/BlobHandler.java:25]: never mind, I dug around in JIRA and can't find it!

**Pull request merged**